src/core/meshcop/dtls.cpp

```diff
@@ -177,6 +199,123 @@ otError Dtls::Start(bool             aClient,
     return MapError(rval);
 }
 
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+
+otError Dtls::StartApplicationCoapSecure(bool             aClient,
+                                         ConnectedHandler aConnectedHandler,
+                                         ReceiveHandler   aReceiveHandler,
+                                         SendHandler      aSendHandler,
+                                         void *           aContext)
+{
+    otExtAddress eui64;
+    int          rval;
+
+    mConnectedHandler      = aConnectedHandler;
+    mReceiveHandler        = aReceiveHandler;
+    mSendHandler           = aSendHandler;
+    mContext               = aContext;
+    mClient                = aClient;
+    mReceiveMessage        = NULL;
+    mMessageSubType        = Message::kSubTypeNone;
+    mApplicationCoapSecure = true;
+
+    if (!mClient)
+    {
+        otLogCritCoap(this, "Application Coap Secure DTLS: No Server functionality implemented!");
+        while (1)
+        {
+            // ToDo: no server functionality implemented.
+        }
+    }
+
+    mbedtls_ssl_init(&mSsl);
+    mbedtls_ssl_config_init(&mConf);
+    mbedtls_ctr_drbg_init(&mCtrDrbg);
+    mbedtls_entropy_init(&mEntropy);
+
+    rval = mbedtls_entropy_add_source(&mEntropy, &Dtls::HandleMbedtlsEntropyPoll, NULL, MBEDTLS_ENTROPY_MIN_PLATFORM,
```

Comment:
A bunch of those code is common with `Dtls::Start()`, can we reduce code duplication here?

Reply:
Yes, of course.
We use the same Function to start the DTLS (Start) as meshcop yet. Done.